### PR TITLE
Change multi instrumentation feature gate into command line flag

### PR DESCRIPTION
**Pull request by changexd**
**Description:** <Describe what has changed.>

Change multi instrumentation feature gate into command line flag

**Link to tracking Issue:** <Issue number if applicable>
Resolves: #2582

**Testing:** <Describe what testing was performed and which tests were added.>
I've changed every `featuregate.EnableMultiInstrumentationSupport` to `config.EnableMultiInstrumentation`

**Documentation:** <Describe the documentation added.>
This is my first attempt to change instrumentation feature gate into command line flags, I wasn't sure if this is how  "normal command line flag"  should look like, feel free to let me know if anything needs to be changed :), then I'll proceed to changing others, also should we group all instrumenation support into a command or separate commands?

### CODE REVIEW

File: .chloggen/multi-instrumentation-command-flag.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: enhancement
```
Comment (pavolloffay) [inline .chloggen/multi-instrumentation-command-flag.yaml:2]: I would prefer to mark it as a braking change

File: .chloggen/multi-instrumentation-command-flag.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+# The name of the component, or a single word describing the area of concern, (e.g. operator, target allocator, github action)
+component: operator
+
+# A brief description of the change. Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: change multi instrumentation feature gate into command line flag
```
Comment (pavolloffay) [inline .chloggen/multi-instrumentation-command-flag.yaml:8]: could you please put here the old feature flag a new cmd flag or put it into the subtext

File: .chloggen/multi-instrumentation-command-flag.yaml (hunk 1 above)
Comment (swiatekm-sumo) [reply .chloggen/multi-instrumentation-command-flag.yaml:2]: It should definitely be a breaking change.

File: main.go [hunk 3]
```diff
@@ -126,6 +127,7 @@ func main() {
 		"Enable leader election for controller manager. "+
 			"Enabling this will ensure there is only one active controller manager.")
 	pflag.BoolVar(&createRBACPermissions, "create-rbac-permissions", false, "Automatically create RBAC permissions needed by the processors")
+	pflag.BoolVar(&enableMultiInstrumentation, "enable-multi-instrumentation", false, "Controls whether the operator supports multi instrumentation")
```
Comment (swiatekm-sumo) [inline main.go:130]: @pavolloffay @jaronoff97 @TylerHelmuth are we ok with this naming convention? Ideally we should be consistent with the flag names, is `--enable-xxx-instrumentation` good?

**Review by swiatekm-sumo — COMMENTED** [review]
The change itself LGTM, just need to make sure we communicate this right to our users.

File: main.go (hunk 3 above)
Comment (TylerHelmuth) [reply main.go:130]: I think it is ok.  The individual language ones could be `enable-xxx-instrumentation`

**Review by pavolloffay — APPROVED** [review]

**Review by jaronoff97 — APPROVED** [review]

**Pull request merged**